lesson_2/calculator.py

```diff
@@ -13,48 +15,61 @@ def invalid_input(str_num):
         return True
     return False
 
-prompt('Welcome to Calculator!')
+def calc_again():
+    prompt('''Would you like to do another calculation?
+        Enter Y to continue OR any other key to stop.''')
+    answer = input()
+    if answer.lower() == 'y':
+        return calculator()
+    return print('See you later!')
 
-# NUM1
-prompt('Enter a number!')
-num1 = input()
-while invalid_input(num1):
-    prompt("Error: That isn't a valid number, expecting an integer!")
+
+def calculator():
+    prompt('Welcome to Calculator!')
+
+    # NUM1
+    prompt('Enter a number!')
     num1 = input()
-num1 = int(num1) # ugly conversion to int, but following tut they did
-# -- def know i can make better
-
-# NUM2
-prompt('Enter another number!')
-num2 = input()
-while invalid_input(num2):
-    prompt("Error: That isn't a valid number, expecting an integer!")
+    while invalid_input(num1):
+        prompt("Error: That isn't a valid number, expecting an integer!")
+        num1 = input()
+    num1 = int(num1)
```

Comment:
Here's how I would put this in a function and reuse it:

```py
def get_number_from_user(prompt_message):
  prompt(prompt_message)
  
  number = input()
  
  while invalid_input(number):
    prompt("Error: That isn't a valid number, expecting an integer!")
    number = input()
  
  return int(number)
```

Then you can do:

```py
num1 = get_number_from_user("Enter a number!")
num2 = get_number_from_user("Enter another number!")
```

Reply:
ahh very nice, will do